modules/miner/cpuminer.go

```diff
@@ -20,8 +27,15 @@ func (m *Miner) threadedMine() {
 	// occurring.
 	cycleStart := time.Now()
 	for {
-		// Kill the thread if mining has been turned off.
 		m.mu.Lock()
+		// Kill the thread if the miner's threadgroup is stopped.
+		if m.tg.IsStopped() {
+			m.miningOn = false
+			m.mining = false
```

Comment:
If we are in the process of closing setting these values to false shouldn't matter, right? Returning immediately would let you put the `IsStopped` check outside of the lock.